File: src/main/java/com/example/cqrs/command/repository/AccountRepository.java
package com.example.cqrs.command.repository;

import com.example.cqrs.command.entity.AccountEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

/**
 * readEntityManagerFactory와 readTransactionManager 사용
 * 패키지 위치에 따라 자동으로 읽기 DB 사용
 */
@Repository
public interface AccountRepository extends JpaRepository<AccountEntity, String> {
}

File: src/main/java/com/example/cqrs/infrastructure/persistence/command/repository/AccountRepository.kt
package com.example.cqrs.infrastructure.persistence.command.repository

import com.example.cqrs.infrastructure.persistence.command.entity.AccountEntity
import org.springframework.data.jpa.repository.JpaRepository

/**
 * 계좌 엔티티 리포지토리
 * 계좌의 현재 상태 관리
 */
interface AccountRepository : JpaRepository<com.example.cqrs.infrastructure.persistence.command.entity.AccountEntity, String> {
    // 필요한 경우 추가 쿼리 메서드를 정의할 수 있음
    // 예: 사용자별 계좌 찾기
    fun findByUserId(userId: String): List<com.example.cqrs.infrastructure.persistence.command.entity.AccountEntity>
}
